src/ameba/rule/lint/unused_instance_variable_access.cr

```diff
@@ -0,0 +1,60 @@
+module Ameba::Rule::Lint
+  # A rule that disallows unused instance variable access.
+  #
+  # For example, this is considered invalid:
+  #
+  # ```
+  # class MyClass
+  #   @my_var : String = "hello"
+  #
+  #   @my_var
+  #
+  #   def hello : String
+  #     @my_var
+  #
+  #     "hello, world!"
+  #   end
+  # end
+  # ```
+  #
+  # And these are considered valid:
+  #
+  # ```
+  # class MyClass
+  #   @my_var : String = "hello"
+  #
+  #   @my_other_var = @my_var
+  #
+  #   def hello : String
+  #     return @my_var if @my_var == "hello"
+  #
+  #     "hello, world!"
+  #   end
+  # end
+  # ```
+  #
+  # YAML configuration example:
+  #
+  # ```
+  # Lint/UnusedInstanceVariableAccess:
+  #   Enabled: true
+  # ```
+  class UnusedInstanceVariableAccess < Base
+    properties do
+      since_version "1.7.0"
+      description "Disallows unused access to instance variables"
+    end
+
+    MSG = "Value from instance variable access is unused"
+
+    def test(source : Source)
+      AST::ImplicitReturnVisitor.new(self, source)
+    end
+
+    def test(source, node : Crystal::InstanceVar, in_macro : Bool)
+      return if in_macro && node.name.in?("@type")
+
+      issue_for node, MSG
```

Comment:
Is this required? `{% @type %}` expression should be flagged too, ain't that right?

```suggestion
      issue_for node, MSG
```

Reply:
Ah yes, I forgot about it, good catch!
I wonder is that's the case with `@def` as well, or just the `@type`...

Reply:
For now maybe it's worth having the ivar / cvar rules not care about vars in macros altogether

Reply:
Idk, it seems that the current implementation holds well enough.
One thing I might change is:

```suggestion
      return if in_macro && node.name == "@type"

      issue_for node, MSG
```